### PR TITLE
move build related files out of the root directory

### PR DESCRIPTION
**Pull request by mikedanese**
Turn the files in root to symlinks into build/root/

alternative to OWNERS changes. what do you guys think of this?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,536 +0,0 @@
-# Copyright 2016 The Kubernetes Authors.
```
Comment (xiangpengzhao) [inline Makefile]: So developers can still run `make foo` in kubernetes root directory, right? I'm not sure if there is something path-related in original Makefile.

Comment (mikedanese) [inline Makefile]: Yes. Everything still works.

**Comment by thockin** [comment]
What will this do if/when we try to have a Windows build?

**Comment by mikedanese** [comment]
I think we can cross that bridge when we get there. We already make use of symlinks e.g. vendor/k8s.io symlinking into staging/

**Pull request merged**